src/index.js

```diff
@@ -57,47 +53,66 @@ async function dispatch({ cmd, data }) {
 async function fetchMetadata(targetUrl) {
  const proxyConfig = await getScrapingConfig();
 
- const {
-  body: html,
-  url,
-  headers,
-  redirectUrls = [],
- } = await got(targetUrl, {
-  timeout: {
-   request: REQUEST_TIMEOUT_MS,
-  },
-  retry: {
-   limit: 0,
-  },
-  ...proxyConfig, // Add proxy configuration here
+ const response = await axios({
+  method: "get",
+  url: targetUrl,
+  timeout: REQUEST_TIMEOUT_MS,
+  responseType: "stream",
+  headers: { Range: "bytes=0-65535" },
+  ...proxyConfig,
+ });
+
+ const contentType = response.headers["content-type"] || "";
+ let html = "";
+ let hasEnded = false;
+
+ return new Promise((resolve, reject) => {
```

Comment:
doesn't axios have this built in?   `maxContentLength`

https://axios-http.com/docs/req_config